kyaml/yaml/rnode.go

```diff
@@ -248,6 +377,619 @@ func (rn *RNode) IsNilOrEmpty() bool {
+	node, found, err := o.nestedValue(fields...)
+	if err != nil || !found {
+		return nil, found, err
+	}
+	if node.value.Kind != yaml.SequenceNode {
+		return nil, found, fmt.Errorf("the YNode kind should be SequenceNode instead of %v", node.value.Kind)
+	}
+	return node, found, nil
+}
+
+func (o *KubeObject) nestedScalar(fields ...string) (*RNode, bool, error) {
+	node, found, err := o.nestedValue(fields...)
+	if err != nil || !found {
+		return nil, found, err
+	}
+	if node.value.Kind != yaml.ScalarNode {
+		return nil, found, fmt.Errorf("the YNode kind should be ScalarNode instead of %v", node.value.Kind)
+	}
+	return node, found, nil
+}
+
+// NestedString returns a nested string.
+func (o *KubeObject) NestedString(fields ...string) (string, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return "", found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagString:
+		return scalarNode.value.Value, true, nil
+	default:
+		return "", true, fmt.Errorf("node was not a string, was %v", scalarNode.value.Tag)
+	}
+}
+
+// NestedInt returns a nested int.
+func (o *KubeObject) NestedInt(fields ...string) (int, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return 0, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagInt:
+		i, err := strconv.Atoi(scalarNode.value.Value)
+		if err != nil {
+			return 0, true, fmt.Errorf("unable to parse %q as int: %w", scalarNode.value.Value, err)
+		}
+		return i, true, nil
+	default:
+		return 0, true, fmt.Errorf("node was not an int, was %v", scalarNode.value.Tag)
+	}
+}
+
+// NestedFloat returns a nested float.
+func (o *KubeObject) NestedFloat(fields ...string) (float64, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return 0, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagFloat:
+		f, err := strconv.ParseFloat(scalarNode.value.Value, 64)
+		if err != nil {
+			return 0, true, fmt.Errorf("unable to parse %q as float: %w", scalarNode.value.Value, err)
+		}
+		return f, true, nil
+	default:
+		return 0, true, fmt.Errorf("node was not a float, was %v", scalarNode.value.Tag)
+	}
+}
+
+// NestedBool returns a nested boolean.
+func (o *KubeObject) NestedBool(fields ...string) (bool, bool, error) {
+	scalarNode, found, err := o.nestedScalar(fields...)
+	if err != nil || !found {
+		return false, found, err
+	}
+
+	switch scalarNode.value.Tag {
+	case NodeTagBool:
+		b, err := strconv.ParseBool(scalarNode.value.Value)
+		if err != nil {
+			return false, true, fmt.Errorf("unable to parse %q as bool: %w", scalarNode.value.Value, err)
+		}
+		return b, true, nil
+	default:
+		return false, true, fmt.Errorf("node was not a bool, was %v", scalarNode.value.Tag)
+	}
+}
+
+// LineComment return the line comment of the field.
+func (o *KubeObject) LineComment(fields ...string) (string, error) {
```

Comment:
qq: Just curious, why is "FootComment" not needed? Is this relate to kpt Setter (like only head and line comments are used)?

Reply:
I don't see any use cases for foot comment. We can add it if we start to see a use case.